src/cli.zig

```diff
@@ -323,6 +365,27 @@ test "parse_size" {
     try expectEqual(@as(u64, 1000 * kib), parse_size("  1000  kb "));
 }
 
+/// Given a limit ilke `10GiB`, return the maximum power-of-two count of `T`s
+/// that can fit in the limit.
+fn parse_size_to_count(comptime T: type, string_opt: ?[]const u8, comptime default: u32) u32 {
+    comptime assert(math.isPowerOfTwo(default));
+
+    var result = default;
+    if (string_opt) |string| {
+        const byte_size = parse_size(string);
+        const count = math.floorPowerOfTwo(u64, byte_size / @sizeOf(T));
+        result = math.cast(u32, count) catch |err| switch (err) {
+            error.Overflow => fatal("size value too large: '{s}'", .{string}),
+        };
```

Comment:
We also need to check that `result >= 2048` — to simplify `SetAssociativeCache` it doesn't support smaller caches than that (2048 comes from the clock algorithm it uses, though I don't remember exactly how...).